packages/editor/src/plugins/geogebra/renderer.tsx

```diff
@@ -1,70 +1,100 @@
 
 export interface GeogebraRendererProps {
   geogebraId: string
-  url: string
 }
 
-export function GeogebraRenderer({ geogebraId, url }: GeogebraRendererProps) {
+// Only re-render if props.geogebraId changes. Reason: Geogebra applet has its own state that is not managed by react. On re-render this state would get lost, meaning the applet would reset.
+export const GeogebraRenderer = memo(
+  GeogebraRenderer_,
+  (previousProps, nextProps) =>
+    previousProps.geogebraId === nextProps.geogebraId
+)
+
+function GeogebraRenderer_({ geogebraId }: GeogebraRendererProps) {
   const [uuid] = useState(uuid_v4())
   const elementId = `ggb-element-${uuid}`
   const containerId = `${elementId}-container`
 
+  // Inject geogebra applet into dom
+  useEffect(() => {
+    const geogebraAvailable = Object.hasOwn(globalThis, 'GGBApplet')
+    if (geogebraAvailable) {
+      injectGeogebra()
+    } else {
+      fetchScriptAndInitializeGeogebra()
+        .then(injectGeogebra)
+        .catch(() =>
+          // eslint-disable-next-line no-console
+          console.error(
+            'Failed to fetch Geogebra script deployggb.js. Geogebra applet will not be visible.'
+          )
+        )
+    }
+
+    function injectGeogebra() {
+      // https://wiki.geogebra.org/en/Reference:GeoGebra_App_Parameters
+      const params = {
+        appName: '…',
+        showToolBar: false,
+        showAlgebraInput: false,
+        showMenuBar: false,
+        material_id: geogebraId,
+        showResetIcon: true,
+        enableLabelDrags: false,
+        enableShiftDragZoom: false,
+        enableRightClick: false,
+        capturingThreshold: null,
+        showToolBarHelp: false,
+        errorDialogsActive: true,
+        useBrowserForJS: false,
+        enableFileFeatures: false,
+        borderColor: 'transparent',
+        scaleContainerClass: containerId,
+      }
+
+      //@ts-expect-error no types for Geogebra script
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+      const applet = new globalThis.GGBApplet(params, true)
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+      applet.inject(elementId)
+    }
+
+    async function fetchScriptAndInitializeGeogebra() {
+      return new Promise<void>((resolve, reject) => {
+        // deployggb.js will define 'GGBApplet' on the global object. We use 'globalThis' instead of 'global' or 'window' for compatibility.
+        // https://javascript.info/global-object
+        // Using 'global' lead to error 'global not defined' in editor package when using vite.
+        const scriptTag = document.createElement('script')
+        scriptTag.src = 'https://www.geogebra.org/apps/deployggb.js'
+        scriptTag.addEventListener('load', () => {
+          resolve()
+        })
+        scriptTag.addEventListener('error', () => {
+          reject()
+        })
```

Comment:
If it's not too much extra work, maybe you could also handle removing these event listeners?

Reply:
Hm, I thought about it but don't think it is necessary. Each time this code is run the listeners are added to a newly created script tag, so it should never happen that we add duplicates. But I might be missing something.